src/index.rs

```diff
@@ -1167,6 +1169,20 @@ impl Index {
         values
     }
 
+    /// Tries to return an enum variant defined within a POU
+    pub fn find_enum_variant_in_pou(&self, pou: &str, variant: &str) -> Option<&VariableIndexEntry> {
+        self.get_enum_variants_in_pou(pou).into_iter().find(|it| it.name == variant)
+    }
+
+    /// Returns all enum variants defined within a POU
+    pub fn get_enum_variants_in_pou(&self, pou: &str) -> Vec<&VariableIndexEntry> {
+        self.enum_qualified_variables
+            .keys()
+            .filter(|key| key.starts_with(&internal_type_name(pou, "").to_lowercase()))
```

Comment:
i think this starts_with is a problem
you may find an enum in a different pou with a similar name

Reply:
i added a failing test

Reply:
ah crap .... this PR is based on another PR ... I think this comment goes better to the other PR ... *args*

Reply:
Thanks for the review, this is such a good edge-case that I couldn't think of; anyways I've cherry-picked the failing test commit into https://github.com/PLC-lang/rusty/pull/1092 (though since this PR is based on the linked PR, it should still be present here)